src/Symfony/Component/Form/Tests/Extension/Core/Type/ChoiceTypeTest.php

```diff
@@ -300,6 +303,27 @@ public function testSubmitSingleNonExpandedEmpty()
 
         $this->assertNull($form->getData());
         $this->assertSame('', $form->getViewData());
+        $this->assertTrue($form->isSynchronized());
+    }
+
+    public function testSubmitSingleNonExpandedEmptyExplicitEmptyChoice()
+    {
+        $form = $this->factory->create('choice', null, array(
+            'multiple' => false,
+            'expanded' => false,
+            'choices' => array(
+                'EMPTY_CHOICE' => 'Empty',
```

Comment:
doesn't this have to use choices_as_values: true to not use the deprecated functionality? I guess the trigger_error is missing for the old choices structure.